src/ir/named.rs

```diff
@@ -327,10 +327,11 @@ impl<'ctx, 'gen> MonotoneFramework for UsedTemplateParameters<'ctx, 'gen>  {
                     &TypeKind::TemplateInstantiation(ref inst) => {
                         let decl = ctx.resolve_type(inst.template_definition());
                         let args = inst.template_arguments();
+                        // Although template definitions should always have
+                        // template parameters, there is a single exception:
+                        // opaque templates. Hence the unwrap_or.
```

Comment:
This ends up being a bit gross to assert because of typedefs having different names than the underlying type, needing to see through resolved type references, and the fact that either of the template instantiation or the template definition can be opaque.

Unless you feel very strongly about it, I'm going to skip this assert for now.